project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -109,15 +105,19 @@ private static void addDocumentForUser(String hash, long userID) {
     Query query = new Query("User").addFilter("__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("User", userID));
     Entity userEntity = getDatastore().prepare(query).asSingleEntity();
     ArrayList<String> docHashes = getUsersDocumentsHashes(userID);
-
     docHashes.add(hash);
-    //user.setDocs(docHashes);
     userEntity.setProperty("documents", docHashes);
+    getDatastore().put(userEntity);
+  }
+
+  public static ArrayList<String> getUsersDocumentsHashes(long userID) {
+    User user = getUserByID(userID);
+    return user.getDocs();
   }
 
+  /* Document Entity */
   public static Document createDocument(String name, String language, String hash, long userID) {
```

Comment:
I believe that this function is still not updating the instance of the User's docHashes property

Reply:
line 131 updates the docHashes property (addDocumentForUser(hash, userID))

Reply:
I saw that. To my understaning this updates the list of the database but not for the instance of the User class that we have already created

Reply:
this should be fine since this function never creates a User object. Since we don't have a case in our codebase where we have created a User, add a document, then use that User object, this should not be an issue. This is because any instance of "adding a document" occurs in this createDocument function.